lib/sycamore/sycamore/transforms/markdown.py

```diff
@@ -0,0 +1,53 @@
+from sycamore.data import Document, Element
+from sycamore.plan_nodes import Node, SingleThreadUser, NonGPUUser
+from sycamore.transforms.map import Map
+from sycamore.utils.markdown import elements_to_markdown
+
+
+class Markdown(SingleThreadUser, NonGPUUser, Map):
+    """
+    The Markdown transform collapses all the elements of the Document
+    into a single Element which is the Markdown representation of the
+    entire Document, as the text_representation.
+
+    Args:
+        child: The source node or component that provides the documents
+        kwargs: Optional resource-related arguments for the operation
+
+    Example:
+        .. code-block:: python
+
+            md = Markdown(child=node)
+            dataset = md.execute()
+    """
+
+    def __init__(self, child: Node, **kwargs):
+        super().__init__(child, f=make_markdown, **kwargs)
+
+
+def make_markdown(doc: Document) -> Document:
+    elems = doc.elements
+    text = elements_to_markdown(elems)
+    pageset: set[int] = set()
+    for elem in elems:
+        pn = elem.properties.get("page_number")
+        if pn is not None:
+            pageset.add(pn)
+    pages = sorted(pageset)
+    if not pages:
+        pages.append(1)
+    doc.elements = [
+        Element(
+            {
+                "type": "Text",
+                "bbox": (0.0, 0.0, 1.0, 1.0),  # best guess
```

Comment:
Why this as opposed to None?

Reply:
It seems there's a fair amount of code that expects to see bounding boxes.  I could leave it unset.  Not sure if there would be any consequences, good or bad.